arch/xtensa/src/esp32/esp32_gpio.c

```diff
@@ -222,12 +222,12 @@ int esp32_configgpio(int pin, gpio_pinattr_t attr)
     }
   else
     {
-      func |= (uint32_t)((2 >> FUNCTION_SHIFT) << MCU_SEL_S);
+      func |= (uint32_t)(PIN_FUNC_GPIO << MCU_SEL_S);
```

Comment:
For ESP32, Function 3 is the GPIO function for all GPIOs.

Reply:
`FUNCTION_3` is just a definition of the ESP32 GPIO API, which configures the field `MCU_SEL` of the respective IO_MUX register with the value `2`.
The IO_MUX API defines `PIN_FUNC_GPIO` to ease the configuration of any pin as GPIO.

Reply:
Ah I see!  it's the shift that's not correct not the `2`.  I thought you just changed the `2` to `0`.  2 is FUNCTION3 and 0 is FUNCTION1

Reply:
Exactly, previously this line always ORed a 0 value because of the shift hehe